### PR TITLE
Add deprecation expectations - fixes #51

### LINKED ISSUE #51: Deprecation expectations
In RxJS, we have a significant number of overload signatures that are deprecated and seemingly unrelated changes to overload signatures and effected false positives for deprecations.

It would be nice to be able to add tests to ensure that deprecations as behaving as they should. To this end, I've [added](https://github.com/ReactiveX/rxjs/pull/5123) a TSLint rule that can be used with dtslint to test our deprecations.

The tests will look something like this:

```ts
it('should deprecate correctly', () => {
  of(queueScheduler); // $ExpectDeprecation
  of(a, queueScheduler); // $ExpectDeprecation
  of(a, b, queueScheduler); // $ExpectDeprecation
  of(a, b, c, queueScheduler); // $ExpectDeprecation
  of(a, b, c, d, queueScheduler); // $ExpectDeprecation
  of(a, b, c, d, e, queueScheduler); // $ExpectDeprecation
  of(a, b, c, d, e, f, queueScheduler); // $ExpectDeprecation
  of(a, b, c, d, e, f, g, queueScheduler); // $ExpectDeprecation
  of(a, b, c, d, e, f, g, h, queueScheduler); // $ExpectDeprecation
  of(a, b, c, d, e, f, g, h, i, queueScheduler); // $ExpectDeprecation  
  of<A>(); // $ExpectDeprecation
  of(); // $ExpectNoDeprecation
  of(a); // $ExpectNoDeprecation
  of(a, b); // $ExpectNoDeprecation
  of(a, b, c); // $ExpectNoDeprecation
  of(a, b, c, d); // $ExpectNoDeprecation
});
```

Would you be open to adding two additional expectations to this package so that the presence and absence of deprecations could be tested?

### PR DESCRIPTION
**Pull request by SamVerschueren**
Want to review it myself as well. But this should resolve #51.

// @sindresorhus @cartant

### CODE REVIEW

File: source/lib/assertions/assert.ts [hunk 1]
```diff
@@ -8,6 +8,26 @@ export const expectType = <T>(value: T) => {		// tslint:disable-line:no-unused
 	// Do nothing, the TypeScript compiler handles this for us
 };
 
+/**
+ * Assert the type of the value.
```
Comment (SamVerschueren) [inline source/lib/assertions/assert.ts]: Update docs

File: source/lib/assertions/assert.ts [hunk 2]
```diff
@@ -8,6 +8,26 @@ export const expectType = <T>(value: T) => {		// tslint:disable-line:no-unused
...
+ *
+ * @param value - Value that should be type checked.
+ */
+// @ts-ignore
+export const expectDeprecated = (value: any) => {		// tslint:disable-line:no-unused
+	// Do nothing, the TypeScript compiler handles this for us
+};
+
+/**
+ * Assert the type of the value.
```
Comment (SamVerschueren) [inline source/lib/assertions/assert.ts]: Fix docs

File: source/lib/assertions/handlers/expect-deprecated.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+			diagnostics.push(makeDiagnostic(node, options.message(message || '?')));
+		}
+
+		return diagnostics;
+	};
+};
+
+/**
+ * That the argument is marked as deprecated.
```
Comment (SamVerschueren) [inline source/lib/assertions/handlers/expect-deprecated.ts]: Fix

File: readme.md [hunk 4]
```diff
@@ -152,6 +152,14 @@ Check if the function call has argument type errors.
 
 Check if a value is of the provided type `T`.
 
+### expectDeprecated(value)
+
+Check that `value` is marked a `@deprecated`.
```
Comment (sindresorhus) [inline readme.md]: Maybe linkify @deprecated to https://jsdoc.app/tags-deprecated.html

**Comment by SamVerschueren** [comment]
Updated. Great tip!

Do we want to add examples in the `readme.md` per assertion? So it's clear what it does. Wanted to keep the [Assertions](https://github.com/SamVerschueren/tsd#assertions) part small because you want to easily see an overview of all the different possibilities.

I find it quite hard to write documentation for larger projects :p.

**Pull request merged**